flink-connector-mongodb-cdc/src/main/java/com/ververica/cdc/connectors/mongodb/source/reader/fetch/MongoDBStreamFetchTask.java

```diff
@@ -117,6 +117,7 @@ public void execute(Context context) throws Exception {
                 SourceRecord changeRecord = null;
                 if (!next.isPresent()) {
                     long untilNext = nextUpdate - time.milliseconds();
+                    nextUpdate += sourceConfig.getPollAwaitTimeMillis();
```

Comment:
I think we should add adjust next update time under 
```
if (heartbeatManager != null) {
                        changeRecord =
                                heartbeatManager
                                        .heartbeat()
                                        .map(this::normalizeHeartbeatRecord)
                                        .orElse(null);
}
nextUpdate = time.milliseconds() + sourceConfig.getPollAwaitTimeMillis();
```

Reply:
Thank you for your guidance. This is really better. Previous settings may result in failure to penalize updated heartbeat。